grease/src/Grease/Macaw/ResolveCall.hs

```diff
@@ -497,6 +500,58 @@ data LookupSyscallResult p sym arch atps rtps where
     Stubs.SomeSyscall p sym (Symbolic.MacawExt arch) ->
     LookupSyscallResult p sym arch atps rtps
 
+newtype SyscallHandleBuilder p sym arch
+  = SyscallHandleBuilder
+      ( forall atps rtps rtp blocks r ctx.
+        ArchContext arch ->
+        Int ->
+        Ctx.Assignment C.TypeRepr atps ->
+        Ctx.Assignment C.TypeRepr rtps ->
+        CS.CrucibleState
+          p
+          sym
+          (Symbolic.MacawExt arch)
+          rtp
+          blocks
+          r
+          ctx ->
+        LookupSyscallResult p sym arch atps rtps
+      )
+
+newtype PlatformContext arch
```

Comment:
Definitely could use a Haddock. In particular, it would be nice to contrast this with `ArchCtx` (i.e., what should go in here, and what should go in `ArchCtx`?)